tools/changelog/README.md

```diff
@@ -0,0 +1,54 @@
+# Generating repository changelogs
+
+This script uses the developer's clone of the repository to generate a changelog via `git log`.
+
+## Requirements:
+- Node.js > 10.0.0
+- Current clone of https://github.com/microsoft/botframework-sdk
+- Current clone of the repository and branch to generate a changelog from
+
+## Preferred setup:
+- **Each repository should only permit squash merges**, otherwise the developer calling the script will need to manually update entries.
+  - Squash merges on GitHub result in commit messages like the following: `"Fix fetch in token server used in E2E Streaming Test (#2944)"`
+  - Other merges will result in commit messages such as: `"Merge pull request #1411 from <fork/branch>"`
+  - This results in a non-descriptive format:
+  ```md
+  - Merge pull request #4661 from microsoft/johtaylo/private-cloud-patch [[PR 4661]](https://github.com/microsoft/botbuilder-dotnet/pull/4661)
+  ```
+- **Developers should generate the changelog from the branch that is about to be released**
+  - Instead of generating it on the default or development branch (e.g `main`), the script should be run on the version branch (e.g. `4.11`).
+
+## Steps
+_The steps below use [botbuilder-dotnet](https://github.com/microsoft/botbuilder-dotnet) as an example._
+- Clone the target repository and switch to the selected branch for generating a changelog.
+  - `git clone -b 4.11 https://github.com/microsoft/botbuilder-dotnet && cd botbuilder-dotnet`
+- Run `git log` with the following flags and arguments.
+  - `git log --show-pulls --pretty=tformat:"%s" --since=2020-08-13 > dotnet.txt`
```

Comment:
Instead of `--since` you could use: `git log $LAST_RELEASE_COMMIT_SHA..` which will just log commits after `$LAST_RELEASE_COMMIT_SHA`.

Reply:
Good feedback, I'll work that into the PR